### ISSUE TITLE
Ordering should be based on `exportKind`

### ISSUE THREAD

**Issue by orokanasaru** [jrdrg/eslint-plugin-sort-exports#23]
TypeScript/Flow have both `export` and `export type` syntax. These are supported by Babel and TypeScript/Flow eslint parsers by adding a field `exportKind` on `NamedModuleDeclaration` that gets set to `value` or `type` as appropriate

In the current implementation of this plugin, both

```ts
export type {Foo};
export {Bar};
```

and 

```ts
export {Foo};
export type {Bar};
```

are flagged as violating this rule, which makes it impossible to have a consistent style throughout the codebase

**Comment by jrdrg** [comment]
Added a `sortExportKindFirst` option in v0.5.0 which can be either `type`, `value`, or `none`, indicating which `exportKind` comes first in the sorted list.
